pkg/services/oauth/oauth.go

```diff
@@ -56,34 +95,196 @@ func (s *service) Configure() (err error) {
 		logrus.Info("Missing OAuth JWT signing key, generating new one")
 		s.crypto.GenerateKeyPair(s.oauthKeyEntity, false)
 	}
+
+	s.consent = nutsConsentClient.NewConsentStoreClient()
+
 	return
 }
 
 // CreateAccessToken extracts the claims out of the request, checks the validity and builds the access token
 func (s *service) CreateAccessToken(request services.CreateAccessTokenRequest) (*services.AccessTokenResult, error) {
-	// extract the JwtBearerToken
-	jwtBearerToken, err := s.parseAndValidateJwtBearerToken(request.RawJwtBearerToken)
-	if err != nil {
+	context := validationContext{
+		rawJwtBearerToken: request.RawJwtBearerToken,
+	}
+
+	// extract the JwtBearerToken, validates according to RFC003 §5.2.1.1
+	// also check if used algorithms are according to spec (ES*** and PS***)
+	// and checks basic validity. Set jwtBearerToken in validationContext
+	if err := s.parseAndValidateJwtBearerToken(&context); err != nil {
 		return nil, fmt.Errorf("jwt bearer token validation failed: %w", err)
 	}
 
-	// Validate the AuthTokenContainer
-	res, err := s.contractValidator.ValidateJwt(jwtBearerToken.AuthTokenContainer, request.VendorIdentifier)
-	if err != nil {
+	// check the maximum validity, according to RFC003 §5.2.1.4
+	if context.jwtBearerToken.ExpiresAt-context.jwtBearerToken.IssuedAt > OauthBearerTokenMaxValidity {
+		return nil, errors.New("JWT validity to long")
+	}
+
+	// check the actor against the registry, according to RFC003 §5.2.1.3
+	// checks signing certificate and sets vendor, actorName in validationContext
+	if err := s.validateIssuer(&context); err != nil {
+		return nil, err
+	}
+
+	// check if client certificate is issued by vendor according to RFC003 §5.2.1.2
+	if err := s.validateClientCertificate(&context, request.ClientCert); err != nil {
+		return nil, err
+	}
+
+	// check if the custodian is registered by this vendor, according to RFC003 §5.2.1.8
+	if err := s.validateSubject(&context); err != nil {
+		return nil, err
+	}
+
+	// Validate the AuthTokenContainer, according to RFC003 §5.2.1.5
+	// todo: request.VendorIdentifier is deprecated, remove in 0.17
+	var err error
+	if context.contractValidationResult, err = s.contractValidator.ValidateJwt(context.jwtBearerToken.AuthTokenContainer, request.VendorIdentifier); err != nil {
 		return nil, fmt.Errorf("identity token validation failed: %w", err)
 	}
-	if res.ValidationResult == services.Invalid {
+	if context.contractValidationResult.ValidationResult == services.Invalid {
 		return nil, fmt.Errorf("identity validation failed")
```

Comment:
no formatting

Reply:
I think this repo doesn't have a code style checker...